dex/src/main/java/io/crate/data/join/CrossJoinBlockNLBatchIterator.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.data.join;
+
+
+import io.crate.breaker.RowAccounting;
+import io.crate.data.BatchIterator;
+import io.crate.data.Row;
+import io.crate.data.UnsafeArrayRow;
+
+import java.util.ArrayList;
+import java.util.function.IntSupplier;
+
+/**
+ * This BatchIterator is used for both CrossJoins and InnerJoins as for the InnerJoins
+ * the joinCondition is tested later on as a filter Projection.
+ *
+ * The Block Nested Loop algorithm is used to implement the cross join.
+ *
+ * <pre>
+ *     // Nested Loop
```

Comment:
Why do we need this here? We have another iterator for the plain NL.

Reply:
Suppose we don't strictly need it anymore. Still this might be useful to differentiate between the normal and the block-wise execution. A link is probably enough.